File: app/src/main/java/com/woilsy/mock/test/api/ApiService2.java
package com.woilsy.mock.test.api;

import com.woilsy.mock.test.MockBean;
import com.woilsy.mock.type.MockInclude;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.POST;

public interface ApiService2 {

    @POST("/request100")
    Observable<MockBean<List<String>>> getData100();

    @MockInclude
    @POST("/request101")
    Observable<MockBean<List<String>>> getData101();
}

File: app/src/main/java/com/woilsy/mock/test/api/ApiService2.kt
package com.woilsy.mock.test.api

import com.woilsy.mock.annotations.MockObj
import kotlinx.coroutines.flow.Flow
import retrofit2.http.GET

@MockObj
interface ApiService2 {

    @GET("/user/method1")
    suspend fun method1(): List<String>

    //不要加suspend 否则会被默认的CallAdapter转换为Call<Flow<String>>
    @GET("/user/method2")
    fun method2(): Flow<String>

}